Bump version to 2.1.0

AltGr opened the pull request.

rjbou commented:
/!\ merge https://github.com/ocaml/opam-rt/pull/57  and update travis script before merging this one

AltGr commented on src/state/opamSwitchState.ml:
```diff
@@ -101,15 +101,15 @@ let load lock_kind gt rt switch =
-      (OpamVersion.nopatch OpamFormatUpgrade.latest_version)
-     <> 0 then
+      (OpamVersion.nopatch OpamFormatUpgrade.v2_0)
+     < 0 then
```
Maybe the value should be exported as `OpamFormatUpgrade.latest_forward_compatible_version` or something more generic like that ?

rjbou replied:
Updated with `latest_compatible_switch_version`

dra27 commented on configure.ac:
```diff
@@ -1,6 +1,6 @@
 dnl The line below must be formatted AC_INIT(opam,VERSION) with no extra spaces
-AC_INIT(opam,2.0.0)
-AC_COPYRIGHT(Copyright 2012-2017 OcamlPro SAS)
+AC_INIT(opam,2.1.0~beta)
```
```suggestion
AC_INIT(opam,2.1.0~alpha)
```
(seems better to have development versions reporting that, until an actual beta is released?). `alpha` is of course better than `dev`, since a `beta` release is then naturally a greater version.

dra27 commented on configure.ac:
```diff
@@ -1,6 +1,6 @@
-AC_INIT(opam,2.0.0)
-AC_COPYRIGHT(Copyright 2012-2017 OcamlPro SAS)
+AC_INIT(opam,2.1.0~beta)
+AC_COPYRIGHT(Copyright 2012-2018 OcamlPro SAS)
```
```suggestion
AC_COPYRIGHT(Copyright 2012-2019 OcamlPro SAS)
```
🙂

dra27 commented on opam-client.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-client.opam:
```diff
@@ -21,8 +21,8 @@ build: [
 depends: [
-  "opam-state" {= "2.0.0"}
-  "opam-solver" {= "2.0.0"}
+  "opam-state" {= "2.1.0~beta"}
```
```suggestion
  "opam-state" {= "2.1.0~alpha"}
```

dra27 commented on opam-client.opam:
```diff
@@ -21,8 +21,8 @@ build: [
-  "opam-state" {= "2.0.0"}
-  "opam-solver" {= "2.0.0"}
+  "opam-state" {= "2.1.0~beta"}
+  "opam-solver" {= "2.1.0~beta"}
```
```suggestion
  "opam-solver" {= "2.1.0~alpha"}
```

dra27 commented on opam-core.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-devel.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-devel.opam:
```diff
@@ -22,7 +22,7 @@ build: [
 build-test: [make "tests"]
 depends: [
-  "opam-client" {= "2.0.0"}
+  "opam-client" {= "2.1.0~beta"}
```
```suggestion
  "opam-client" {= "2.1.0~alpha"}
```

dra27 commented on opam-format.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-format.opam:
```diff
@@ -21,7 +21,7 @@ build: [
 ]
 depends: [
-  "opam-core" {= "2.0.0"}
+  "opam-core" {= "2.1.0~beta"}
```
```suggestion
  "opam-core" {= "2.1.0~alpha"}
```

dra27 commented on opam-installer.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-installer.opam:
```diff
@@ -21,7 +21,7 @@ build: [
 ]
 depends: [
-  "opam-format" {= "2.0.0"}
+  "opam-format" {= "2.1.0~beta"}
```
```suggestion
  "opam-format" {= "2.1.0~alpha"}
```

dra27 commented on opam-repository.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-repository.opam:
```diff
@@ -21,7 +21,7 @@ build: [
 ]
 depends: [
-  "opam-format" {= "2.0.0"}
+  "opam-format" {= "2.1.0~beta"}
```
```suggestion
  "opam-format" {= "2.1.0~alpha"}
```

dra27 commented on opam-solver.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-format.opam:
```diff
@@ -21,7 +21,7 @@ build: [
 depends: [
-  "opam-core" {= "2.0.0"}
+  "opam-core" {= "2.1.0~beta"}
   "opam-file-format" {>= "2.0.0~rc2"}
```
Doesn't this want updating, while we're here?

dra27 commented on opam-solver.opam:
```diff
@@ -21,7 +21,7 @@ build: [
 ]
 depends: [
-  "opam-format" {= "2.0.0"}
+  "opam-format" {= "2.1.0~beta"}
```
```suggestion
  "opam-format" {= "2.1.0~alpha"}
```

dra27 commented on opam-state.opam:
```diff
@@ -1,5 +1,5 @@
 opam-version: "1.2"
-version: "2.0.0"
+version: "2.1.0~beta"
```
```suggestion
version: "2.1.0~alpha"
```

dra27 commented on opam-state.opam:
```diff
@@ -21,7 +21,7 @@ build: [
 ]
 depends: [
-  "opam-repository" {= "2.0.0"}
+  "opam-repository" {= "2.1.0~beta"}
```
```suggestion
  "opam-repository" {= "2.1.0~alpha"}
```

AltGr commented:
This just got bigger: fixed some corner-cases, and added per-file format version strings. This means that:
- format versions are no longer bound to opam versions (although the field is still named `opam-version:` :grimacing: )
- different file types can be updated at their own pace
- no unnecessary updates just to bump version numbers (that then trigger compat problems)
- probably more bugs since it's an invasive change :innocent:
